src/app/services/user-search.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable, OnDestroy, Inject  } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { DataStore } from './data.store';
+import { AUTH_API_URL } from 'ngx-login-client';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class UserSearchService implements OnDestroy {
+
+  private apiResult; // stores the http observable returned from the api
+  private resultSubscription: Subscription; // stores the subscription to apiResult
+
+  constructor(
+    private http: HttpClient,
+    private savedata: DataStore,
+    @Inject(AUTH_API_URL) private authApiUrl: string
+    ) {
+  }
+
+  public getusers(username: string = null) {
+    if (username != null) {
```

Comment:
It will not be username since it will can just be half name that user will search. Something like `ro` for `rohitkrai`. It sshould be named `searchTerms`